src/python/pants/engine/fs.py

```diff
@@ -117,6 +117,10 @@ def file_stats(self):
     return [p.stat for p in self.files]
 
 
+class MergeDirectoriesRequest(datatype([('directories', tuple)])):
```

Comment:
Naming nit: the "request" bit makes more sense (imo) for process executions, because it is hard to pretend that they are referentially transparent. But for something that is definitely deterministic like snapshot merging, it feels easier to say that we want "the snapshot corresponding to these merged directories". So I'd drop `Request` and make this more nouny... `DirectoryMerger` `MergedDirectories`, etc.

Reply:
Changed to `MergedDirectories`